### PR TITLE
Resharing in e2e test

### PR DESCRIPTION
**Pull request by ScharfViktor**
- [x]  added api methods (for ocis and for oc10): creating group, adding user to group, deleting group

- [x]  created "reshare with user" step

- [ ]  created "reshare with group" step (I'll do it in separate task)
- [ ]  created "share with group" step (I'll do it in separate task)

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>e2e-tests oCIS</strong> https://drone.owncloud.com/owncloud/web/27233/12/1
<details><summary>:boom: To see the trace, please open the link in the console ...</summary>

<p>


#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27233/tracing/folder-alice-2022-8-1-02-00-46.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27233/tracing/folder-alice-2022-8-1-02-02-07.zip 

#### npx playwright show-trace https://cache.owncloud.com/public/owncloud/web/27233/tracing/folder-brian-2022-8-1-02-02-16.zip 


</p></details>

**Comment by ScharfViktor** [comment]
test failed in this place: https://github.com/owncloud/web/blob/master/tests/e2e/support/objects/app-files/utils/sidebar.ts#L50-L53 when I try to transition in sidebar from details-> to shares

https://user-images.githubusercontent.com/84779829/182256010-9531da56-5233-4d52-96fb-bdfd0441f7df.mov



transition or animation different between Shares and Personal
@fschade could you look please what is problem

**Comment by ScharfViktor** [comment]
> test failed in this place: https://github.com/owncloud/web/blob/master/tests/e2e/support/objects/app-files/utils/sidebar.ts#L50-L53 when I try to transition in sidebar from details-> to shares
> 
>  Screen.Recording.2022-08-02.at.00.25.48.mov 
> transition or animation different between Shares and Personal @fschade could you look please what is problem

it works correct without promise: 
```
await panelSelector.click()
await locatorUtils.waitForEvent(nextPanel, 'transitionend')

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature [hunk 1]
```diff
@@ -0,0 +1,46 @@
+Feature: reshare
+
+  Background:
+    And "Admin" disables share auto accepting
```
Comment (SagarGi) [inline tests/e2e/cucumber/features/integrations/reshare.ocis.feature:4]:
```suggestion
    Given "Admin" has disabled share auto accepting
```

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      | Brian |
+      | Carol |
+    And "Admin" creates following group
+      | id      |
+      | sales   |
+      | finance |
+    When "Admin" adds user to the group
+      | user  | group |
+      | Carol | sales |
+    When "Alice" logs in
```
Comment (SwikritiT) [inline tests/e2e/cucumber/features/integrations/reshare.ocis.feature:19]:
```suggestion
    And "Alice" logs in
```

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    And "Brian" reshares the following resource
+      | resource         | user  | role   |
+      | folder_to_shared | Carol | viewer |
+    When "Carol" logs in
+    And "Carol" opens the "files" app
+    And "Carol" navigates to the shared with me page
+    And "Carol" accepts the following share
+      | name             |
+      | folder_to_shared |
+    And "Carol" logs out
```
Comment (SwikritiT) [inline tests/e2e/cucumber/features/integrations/reshare.ocis.feature:43]:
```suggestion
    Then the following users should be displayed in the collaborator list of folder "folder_to_shared" for "Alice"
    |user|role|
    |Brian|editor|
    |Carol|viewer|
    And "Carol" logs out
```
Can we add a step like this here?

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature [hunk 4]
```diff
@@ -0,0 +1,46 @@
+Feature: reshare
+
+  Background:
+    And "Admin" disables share auto accepting
+
```
Comment (SwikritiT) [inline tests/e2e/cucumber/features/integrations/reshare.ocis.feature:5]: I noticed the name of the file is `reshare.ocis.feature` if it's only going to be run for ocis I don't think we need this line? Is auto accept configurable in OCIS? Maybe we can run this test for both backend. `reshare.feature` ?
```suggestion
    Given "Admin" disables share auto accepting

```

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature (hunk 1 above)
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/e2e/cucumber/features/integrations/reshare.ocis.feature:4]: it wasn't implement in this PR. In all e2e tests we do not have given steps

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature (hunk 3 above)
Comment (ScharfViktor) [reply tests/e2e/cucumber/features/integrations/reshare.ocis.feature:43]: We had already used the 'then' steps, but then removed them (not my decision)
`Then the following users should be displayed in the collaborator list of folder "folder_to_shared" for "Alice"` steps partly checked in `And "Carol" accepts the following share` 
I sticked to this style

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature (hunk 4 above)
Comment (ScharfViktor) [reply tests/e2e/cucumber/features/integrations/reshare.ocis.feature:5]: Thanks, I implemented api for both backends, but forgot to change file name

**Comment by ScharfViktor** [comment]
https://drone.owncloud.com/owncloud/web/27317/11/18
https://drone.owncloud.com/owncloud/web/27315/11/18

re-share tests has failed in CI. this needs to be investigated. I'll do it in new issue. In the meantime, let these tests only for ocis

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    
+    And "Carol" logs in
+    And "Carol" opens the "files" app
+    And "Carol" navigates to the shared with me page
+    And "Carol" accepts the following share
+      | name             |
+      | folder_to_shared |
+    And "Carol" reshares the following resource
+      | resource         | user  | role   |
+      | folder_to_shared | Alice | viewer |
```
Comment (SwikritiT) [inline tests/e2e/cucumber/features/integrations/reshare.ocis.feature:50]: I didn't know it was possible to reshare the resource to the person to who the resource belongs with a different roles. Is this expected?

I tried to play around with the UI 
1. Alice accepts the share request
2. Alice as the owner of the resource creates a sub folder inside `folder_to_shared` like `test`
3. The sub-folder is displayed in shared with me inside `folder_to_shared`
4. If Alice goes inside folder_to_shared->tests from shared with me (this folder is shared with role viewer), she can create resources inside it but not directly inside `folder_to_shared`

Probably not related to this PR. Seems like a bug.

Comment (ScharfViktor) [reply tests/e2e/cucumber/features/integrations/reshare.ocis.feature:50]: I didn't try to check this case. It's too complicated. It doesn't seem realistic.
I checked only that user can re-share re-shared resource. I did not create the 4th user

Comment (SwikritiT) [reply tests/e2e/cucumber/features/integrations/reshare.ocis.feature:50]: I'll create a separate issue for this case

**Review by SwikritiT — APPROVED** [review]
LGTM

File: tests/e2e/cucumber/features/integrations/reshare.ocis.feature (hunk 5 above)
Comment (SwikritiT) [reply tests/e2e/cucumber/features/integrations/reshare.ocis.feature:50]: issue created here https://github.com/owncloud/ocis/issues/4336

**Pull request merged**